pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,269 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	admv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	issuerName        = "selfsigned-issuer"
+	certificateName   = "rook-admission-controller-cert"
+	webhookConfigName = "rook-ceph-webhook"
+)
+
+func createIssuer(ctx context.Context, webhookContext *WebhookContext) (*api.Issuer, error) {
+	logger.Infof("Fetching Issuer %s/%s.", namespace, issuerName)
+	issuer, err := webhookContext.Admission.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Infof("Creating Issuer %s/%s.", namespace, issuerName)
+			issuer = &api.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      issuerName,
+					Namespace: namespace,
+				},
+				Spec: api.IssuerSpec{
+					IssuerConfig: api.IssuerConfig{
+						SelfSigned: &api.SelfSignedIssuer{},
+					},
+				},
+			}
+			issuers, _ := webhookContext.Admission.Issuers(namespace).Create(ctx, issuer, metav1.CreateOptions{})
+			sleepTime := 2
+
+			for j := range issuers.Status.Conditions {
```

Comment:
Is the purpose of this loop so we make sure to wait until it's created so the secret will be found later? 
The loop needs to be fixed so that it will retrieve the latest conditions again. Currently, it's not retrieving the latest status with each iteration.

Reply:
nit: you can use the condition type instead of the index to simplify a bit
```suggestion
			for _, condition := range issuers.Status.Conditions {
				if condition.Status == v1.ConditionStatus(api.IssuerConditionReady) {
```

Reply:
I have changed the loop with something different, I have tested that works well now.